protocols/kad/src/behaviour/test.rs

```diff
@@ -1287,3 +1287,87 @@ fn network_behaviour_inject_address_change() {
         kademlia.addresses_of_peer(&remote_peer_id),
     );
 }
+
+#[test]
+fn client_mode() {
+    let mut cfg = KademliaConfig::default();
+    cfg.set_mode(Mode::Client);
+
+    // Create a server and client peer.
+    let client_swarm = build_node_with_config(cfg);
+    let server_swarm = build_node();
+
+    let mut swarms = Vec::new();
+    swarms.push(client_swarm.1);
+    swarms.push(server_swarm.1);
+
+    let mut swarm_addrs = Vec::new();
+    swarm_addrs.push(client_swarm.0);
+    swarm_addrs.push(server_swarm.0);
+
+    let swarm_ids: Vec<_> = swarms
+        .iter()
+        .map(|swarm| swarm.local_peer_id())
+        .cloned()
+        .collect();
+
+    block_on(poll_fn(move |ctx| {
```

Comment:
Wouldn't you need to have the client dial the server before starting the test logic?